### PR TITLE
docs(server): http_proxy example

### PR DESCRIPTION
**Pull request by euclidr**
I added a CONNECT tunneling example #1759

### CODE REVIEW

File: examples/http_proxy.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            if Method::CONNECT == req.method() {
+                // Recieved an HTTP request like:
+                // ```
+                // CONNECT www.domain.com:443 HTTP/1.1
+                // Host: www.domain.com:443
+                // Proxy-Connection: Keep-Alive
+                // ```
+                //
+                // When HTTP method is CONNECT we should return an empty body
+                // then we can eventually upgrade the connection and talk a new protocal.
```
Comment (ahmedcharles) [inline examples/http_proxy.rs]:
```suggestion
                // then we can eventually upgrade the connection and talk a new protocol.
```

**Comment by euclidr** [comment]
@ahmedcharles there have been a lot of change since this example is written. `split` method of `AsyncRead` `AsyncWrite` in tokio-io is temporary removed, which make it hard to implement tunnel. 

I found tokio has plan to get `split` back, [see the issue](https://github.com/tokio-rs/tokio/issues/1256). I will fix this example once the `split` is back. Or are there better ways to implement tunnel?

#1884

**Comment by ahmedcharles** [comment]
See: https://github.com/ahmedcharles/hyper/commit/cc37f87d6ebfd5a3db618852fc1788b2c88db515

It's hacky but it works. (Though, h2 doesn't build at the moment, so when I rebased, it failed to build.)

**Comment by euclidr** [comment]
@ahmedcharles thanks, it works! I changed your code a little, `use tokio_tcp` to `use tokio_net::tcp`, the example runs correctly.

**Comment by ahmedcharles** [comment]
No problem.

**Comment by seanmonstar** [comment]
Thanks for doing this, and sorry for taking so long, I've rebased and merged this into master manually!

**Pull request closed without merging**